documentation/how-to/authorization.md

```diff
@@ -0,0 +1,262 @@
+a client to add the necessary tuples in the following section.
+
+```bash
+juju show-unit temporal-k8s/0
+
+# Output:
+temporal-k8s/0:
+  opened-ports: []
+  charm: local:jammy/temporal-k8s-6
+  leader: true
+  life: alive
+  relation-info:
+    ...
+      openfga: '{"store_id": "<store_id>", "token": "<token>",
+        "address": "10.152.183.144", "port": "8080", "scheme": "http", "auth_model_id": "<model_id>"}'
+```
+
+At this point, our Temporal Server is active again with authorization enabled.
+This means that any request made to the Temporal Server will need to be
+[authenticted](./authentication.md) using Google OAuth.
+
+## Create OpenFGA Authorization Rules
+
+Once authorization is enabled, you will not be able to access any namespace in
+Temporal. You must now create the necessary tuples in the OpenFGA store to
+enable users read/write access to different namespaces.
+
+The authorization model used is as follows:
+
+```
+model
+  schema 1.1
+
+type user
+type group
+  relations
+    define member: [user]
+type namespace
+  relations
+    define admin: [group#member]
+    define reader: [group#member] or writer
+    define writer: [group#member]
+```
+
+The above model is a simple way of relating users to groups to namespaces. A
+`user` can be related to a `group` as a `member`, and a `group` can be related
+to a `namespace` as either a `reader`, `writer` or `admin`.For example, If user
+`john` is a member of group `abc`, and group `abc` is related to namespace
+`example` as a `writer`, then user `john` will be assigned write access on
+namespace `example`.
+
+To add tuples to our OpenFGA store, we can use an
+[OFGA client](https://github.com/canonical/ofga) as follows:
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/ofga"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Create a new ofga client
+	client, err := ofga.NewClient(ctx, ofga.OpenFGAParams{
+		Scheme:      "http", // defaults to `https` if not specified.
+		Host:        "10.152.183.144",
+		Port:        "8080",
+		Token:       "<token>",
+		StoreID:     "<store_id>",
+		AuthModelID: "<model_id>",
+	})
+	if err != nil {
+		// Handle error
+		fmt.Printf("Error connecting to OpenFGA store: %v \n", err)
+	}
+
+    // Add your email as a member of group 'test'
+	err = client.AddRelation(ctx, ofga.Tuple{
+		Object:   &ofga.Entity{Kind: "user", ID: "<your_email>"},
+		Relation: "member",
+		Target:   &ofga.Entity{Kind: "group", ID: "test"},
+	})
+	if err != nil {
+		// Handle error
+		fmt.Printf("Error adding relations: %v", err)
+	}
+
+    // Assign all members of group 'test' write access to the 'default' namespace in Temporal.
+	err = client.AddRelation(ctx, ofga.Tuple{
+		Object:   &ofga.Entity{Kind: "group", ID: "test#member"},
+		Relation: "writer",
+		Target:   &ofga.Entity{Kind: "namespace", ID: "default"},
+	})
+	if err != nil {
+		// Handle error
+		fmt.Printf("Error adding relations: %v", err)
+	}
+}
+```
```

Comment:
This is not Temporal charm specific and is piercing the abstraction of using OpenFGA here. We either provide a Temoral auth cli interface on top of the generic tuple addition OpenFGA provides that doesn't use OpenFGA's language (i.e. `adduser` or `addgroup`) or completely drop this from the documentation. I can see this relevant in the OpenFGA charm documentation.

Reply:
I agree with your perspective. The way I see it, there are two ways we can approach this to avoid having the user run a program or use a cli tool to add tuples:

1. Add actions to the Temporal charm to add, remove and list tuples. These actions would be specific to the authorization model of Temporal (i.e. you specify user, group and namespace for tuples).
2. Add actions to the OpenFGA charm to add, remove and list tuples. These actions would be generic and would work across different authorization models for multiple applications. You would, however, need a way of specifying the store that you are querying.

These two approaches are not mutually exclusive, so I might be leaning towards the first option to have a Temporal-specific wrapper around these actions for now, while also looking into adding actions to the OpenFGA charm to make this use-case more general for the future.

What do you think?

Reply:
To me the the second options seems more appropriate. I don't think temporal should care about adding users and groups (not even as a proxy). We can imagine that the authorization was done centrally and we had no write access to it (which i think it is a common case). You could point to the documentation of the "authorization backend" to add users/permissions.

Reply:
I've refactored this PR and implemented the necessary actions to achieve this. However, it is slightly different to what you described, as OpenFGA deals in tuples, so you cannot add a user or add a group, but rather add the relation between a user and a group. All mentioned operations can be achieved using the four actions that have been implemented.